Tests/SonarScanner.MSBuild.PreProcessor.Test/ArgumentProcessorTests.cs

```diff
@@ -138,6 +138,85 @@ public void PreArgProc_ApiBaseUrl_NotSet_SonarQubeDefault(string hostUri, string
+    }
+
+    [DataTestMethod]
+    [DataRow(" ")]
+    [DataRow("eu")]
+    [DataRow("default")]
+    [DataRow("global")]
+    [DataRow(@"""us""")]
+    public void PreArgProc_Region_Unknown(string region)
+    {
+        var logger = CheckProcessingFails("/k:key", $"/d:sonar.region={region}");
+        logger.AssertErrorLogged($"Unsupported region '{region}'. List of supported regions: 'us'. Please check the 'sonar.region' property.");
+    }
+
+    [TestMethod]
+    public void PreArgProc_Region_Invalid()
+    {
+        var logger = CheckProcessingFails("/k:key", $"/d:sonar.region=");
+        logger.AssertErrorLogged("The format of the analysis property sonar.region= is invalid");
+    }
+
+    [DataTestMethod]
+    [DataRow("us", null, null, null, typeof(CloudHostInfo), "https://sonarqube.us", "https://api.sonarqube.us")]
+    [DataRow("us", null, "https://cloud", "https://api", typeof(CloudHostInfo), "https://cloud", "https://api")]
+    [DataRow("us", null, "https://cloud", null, typeof(CloudHostInfo), "https://cloud", "https://api.sonarqube.us")]
+    [DataRow("us", "https://cloud", "https://cloud", "https://api", typeof(CloudHostInfo), "https://cloud", "https://api", "The arguments 'sonar.host.url' and 'sonar.scanner.sonarcloudUrl' are both set. Please set only 'sonar.scanner.sonarcloudUrl'.")]
+    [DataRow("us", "https://host", null, "https://api", typeof(ServerHostInfo), "https://host", "https://api")]
+    [DataRow("us", "https://host", null, null, typeof(ServerHostInfo), "https://host", "https://host/api/v2")]
+    [DataRow(null, null, null, null, typeof(CloudHostInfo), "https://sonarcloud.io", "https://api.sonarcloud.io")]
+    [DataRow(null, null, "https://cloud", "https://api", typeof(CloudHostInfo), "https://cloud", "https://api")]
+    [DataRow(null, null, "https://cloud", null, typeof(CloudHostInfo), "https://cloud", "https://api.sonarcloud.io")]
+    [DataRow(null, "https://cloud", "https://cloud", "https://api", typeof(CloudHostInfo), "https://cloud", "https://api", "The arguments 'sonar.host.url' and 'sonar.scanner.sonarcloudUrl' are both set. Please set only 'sonar.scanner.sonarcloudUrl'.")]
+    [DataRow(null, "https://host", null, "https://api", typeof(ServerHostInfo), "https://host", "https://api")]
+    [DataRow(null, "https://host", null, null, typeof(ServerHostInfo), "https://host", "https://host/api/v2")]
+    public void PreArgProc_Region_Overrides(string region, string hostOverride, string sonarClourUrlOverride, string apiOverride, Type expectedHostInforType, string expectedHostUri, string expectedApiUri, params string[] expectedWarnings)
+    {
+        var logger = new TestLogger();
+        var args = CheckProcessingSucceeds(
+            logger,
+            Substitute.For<IFileWrapper>(),
+            Substitute.For<IDirectoryWrapper>(),
+            [
+                "/k:key",
+                .. region is null ? Array.Empty<string>() : [$"/d:sonar.region={region}"],
```

Comment:
Any reason not using the property value here?
```suggestion
                .. region is null ? Array.Empty<string>() : [$"/d:{SonarProperties.Region}={region}"],
```